cfme/physical/physical_storage.py

```diff
@@ -0,0 +1,227 @@
+            data_dict['vendor'] = quad_data.xpath(self.QUADRANT.format(pos="c"))[0].get('alt')
+            data_dict['health_state'] = quad_data.xpath(self.QUADRANT.format(pos="d"))[0].get('alt')
+        return data_dict
+
+
+class PhysicalStorageDetailsToolbar(View):
+    """Represents physical toolbar and its controls."""
+    configuration = Dropdown(text="Configuration")
+
+
+class PhysicalStorageDetailsEntities(View):
+    """Represents Details page."""
+    properties = SummaryTable(title="Properties")
+    management_networks = SummaryTable(title="Management Networks")
+    relationships = SummaryTable(title="Relationships")
+    power_management = SummaryTable(title="Power Management")
+    firmwares = SummaryTable(title="Firmwares")
+    ports = SummaryTable(title="Ports")
+
+
+class PhysicalStorageDetailsView(ComputePhysicalInfrastructureStoragesView):
+    """Main PhysicalStorage details page."""
+    breadcrumb = BreadCrumb()
+    toolbar = View.nested(PhysicalStorageDetailsToolbar)
+    entities = View.nested(PhysicalStorageDetailsEntities)
+
+    @property
+    def is_displayed(self):
+        title = "{name} (Summary)".format(name=self.context["object"].name)
+        return (self.in_compute_physical_infrastructure_storages and
+                self.breadcrumb.active_location == title)
+
+
+class PhysicalStoragesToolbar(View):
+    """Represents hosts toolbar and its controls."""
+    configuration = Dropdown(text="Configuration")
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class PhysicalStorageSideBar(View):
+    """Represents left side bar. It usually contains navigation, filters, etc."""
+
+    @View.nested
+    class filters(Accordion): # noqa
+        tree = ManageIQTree()
+
+
+class PhysicalStorageEntitiesView(BaseEntitiesView):
+    """Represents the physical storage view"""
+    @property
+    def entity_class(self):
+        return PhysicalStorageEntity
+
+
+class PhysicalStoragesView(ComputePhysicalInfrastructureStoragesView):
+    toolbar = View.nested(PhysicalStoragesToolbar)
+    sidebar = View.nested(PhysicalStorageSideBar)
+    including_entities = View.include(PhysicalStorageEntitiesView, use_parent=True)
+
+    @property
+    def is_displayed(self):
+        return (self.in_compute_physical_infrastructure_storages and
+                self.title.text == "Physical Storages")
+
+
+@attr.s
+class PhysicalStorage(BaseEntity, Updateable, Pretty, PolicyProfileAssignable, Taggable):
+    """Model of an Physical Storage in cfme.
+
+    Args:
+        name: Name of the physical storage.
+
+    Usage:
+        mystorage = PhysicalStorage(name='sample_Storage')
+        mystorage.create()
+
+    """
+    pretty_attrs = ['name']
+
+    name = attr.ib()
+    provider = attr.ib()
+
+    INVENTORY_TO_MATCH = ['Product Name', 'Serial Number', 'Description',
+                          'Health State', 'Drive Bays', 'Enclosure Count']
+
+    def get_value_from_summary(self, value):
```

Comment:
@mshriver I added this as a separate method for code clarity and so we'll not have to change behavior if we add others functionalities which retrieves attributes from others sources. Don't know if this name is good enough though.